impl/src/test/java/io/kroxylicious/testing/kafka/common/KafkaClusterConfigTest.java

```diff
@@ -39,13 +41,89 @@ void generateConfigForSpecificBroker() {
+        // Given
+        var numBrokers = 3;
+        final KafkaClusterConfig kafkaClusterConfig = kafkaClusterConfigBuilder.kraftMode(true).kraftControllers(3).brokersNum(numBrokers).build();
+
+        // When
+        for (int nodeId = 0; nodeId < numBrokers; nodeId++) {
+            assertNodeIdHasRole(kafkaClusterConfig, nodeId, "broker,controller");
+        }
+    }
+
+    @Test
+    void shouldConfigureMultipleControllersInControllerOnlyMode() {
+        // Given
+        var numBrokers = 1;
+        var numControllers = 3;
+        final KafkaClusterConfig kafkaClusterConfig = kafkaClusterConfigBuilder.kraftMode(true).kraftControllers(numControllers).brokersNum(numBrokers).build();
+
+        // When
+        assertNodeIdHasRole(kafkaClusterConfig, 0, "broker,controller");
+
+        for (int nodeId = 1; nodeId < numControllers; nodeId++) {
+            assertNodeIdHasRole(kafkaClusterConfig, nodeId, "controller");
+        }
+    }
+
+    @Test
+    void shouldConfigureSingleControllersInCombinedMode() {
+        // Given
+        var numBrokers = 3;
+        var numControllers = 1;
+        final KafkaClusterConfig kafkaClusterConfig = kafkaClusterConfigBuilder.kraftMode(true).kraftControllers(numControllers).brokersNum(numBrokers).build();
+
+        // When
+        assertNodeIdHasRole(kafkaClusterConfig, 0, "broker,controller");
+
+        for (int nodeId = 1; nodeId < numBrokers; nodeId++) {
+            assertNodeIdHasRole(kafkaClusterConfig, nodeId, "broker");
+        }
+    }
+
+    @Test
+    void shouldGenerateConfigForBrokerNodes() {
+        // Given
+        var numBrokers = 3;
+        var numControllers = 1;
+        final KafkaClusterConfig kafkaClusterConfig = kafkaClusterConfigBuilder.kraftMode(true).kraftControllers(numControllers).brokersNum(numBrokers).build();
+
+        // When
+        final Stream<KafkaClusterConfig.ConfigHolder> brokerConfigs = kafkaClusterConfig.getBrokerConfigs(() -> endpointConfig);
+
+        // Then
+        assertThat(brokerConfigs).hasSize(3);
+    }
+
+    @Test
+    void shouldGenerateConfigForControllerNodes() {
+        // Given
+        var numBrokers = 1;
+        var numControllers = 3;
+        final KafkaClusterConfig kafkaClusterConfig = kafkaClusterConfigBuilder.kraftMode(true).kraftControllers(numControllers).brokersNum(numBrokers).build();
+
+        // When
+        final Stream<KafkaClusterConfig.ConfigHolder> brokerConfigs = kafkaClusterConfig.getBrokerConfigs(() -> endpointConfig);
+
+        // Then
+        assertThat(brokerConfigs).hasSize(3);
+    }
+
+    private void assertNodeIdHasRole(KafkaClusterConfig kafkaClusterConfig, int nodeId, String expectedRole) {
+        final var config = kafkaClusterConfig.generateConfigForSpecificNode(endpointConfig, nodeId);
+        assertThat(config.getProperties()).extracting(brokerConfig -> brokerConfig.get("process.roles")).as("nodeId: %s to have process.roles", nodeId).isEqualTo(
```

Comment:
We should switch to using the kafka enums throughout in another PR.

Reply:
Ohh that is a good thought